src/sycl/fp8_scaled_mm.cpp

```diff
@@ -0,0 +1,412 @@
+
+  using EpilogueOp = cutlass::epilogue::fusion::LinearCombination<
+      ElementOutput,
+      ElementComputeEpilogue,
+      ElementAccumulator,
+      ElementAccumulator,
+      cutlass::FloatRoundStyle::round_to_nearest>;
+
+  using FusionCallBacks = cutlass::epilogue::fusion::
+      FusionCallbacks<EpilogueDispatchPolicy, EpilogueOp, TileShape, decltype(tile_shape(TiledMma()))>;
+
+  // Use U16 store for FP16/BF16
+  using GmemTiledCopyStore = XE_2D_U16x8x16_ST_N;
+
+  using CollectiveEpilogue = cutlass::epilogue::collective::CollectiveEpilogue<
+      EpilogueDispatchPolicy,
+      TileShape,
+      ElementAccumulator,
+      cutlass::gemm::TagToStrideC_t<LayoutC>,
+      ElementOutput,
+      cutlass::gemm::TagToStrideC_t<LayoutD>,
+      FusionCallBacks,
+      XE_2D_U32x8x16_LD_N,
+      void,
+      void,
+      GmemTiledCopyStore,
+      void,
+      void>;
+
+  using CollectiveMainloop = cutlass::gemm::collective::CollectiveMma<
+      GEMMDispatchPolicy,
+      TileShape,
+      cute::tuple<ElementInputA, ElementScale, StrideScale>,
+      cutlass::gemm::TagToStrideA_t<LayoutA>,
+      cute::tuple<ElementInputB, ElementScale, StrideScale>,
+      cutlass::gemm::TagToStrideB_t<LayoutB>,
+      TiledMma,
+      GmemTiledCopyA,
+      void,
+      void,
+      cute::identity,
+      GmemTiledCopyB,
+      void,
+      void,
+      cute::identity>;
+
+  using GemmKernel =
+      cutlass::gemm::kernel::GemmUniversal<Shape<int, int, int, int>, CollectiveMainloop, CollectiveEpilogue>;
+
+  using Gemm = cutlass::gemm::device::GemmUniversalAdapter<GemmKernel>;
+};
+
+// Helper to check if output is FP8
+static inline bool is_fp8_dtype(at::ScalarType dtype) {
+  return dtype == at::ScalarType::Float8_e4m3fn || dtype == at::ScalarType::Float8_e5m2;
+}
+
+// Helper function to get FP8 min/max values
+static inline std::pair<float, float> get_fp8_range(at::ScalarType dtype) {
+  if (dtype == at::ScalarType::Float8_e4m3fn) {
+    // E4M3FN: max = 448, min = -448
+    return {-448.0f, 448.0f};
+  } else {
+    // Float8_e5m2
+    // E5M2: max = 57344, min = -57344
+    return {-57344.0f, 57344.0f};
+  }
+}
+
+// Helper function to dispatch based on input FP8 type and output dtype
+template <typename ElementInputFp8>
+static at::Tensor fp8_scaled_mm_impl(
+    const at::Tensor& mat_a,
+    const at::Tensor& mat_b,
+    const at::Tensor& scales_a_half,
+    const at::Tensor& scales_b_half,
+    const at::ScalarType out_dtype,
+    at::Tensor& out,
+    const cutlass::KernelHardwareInfo& hw_info) {
+  at::Tensor mat_a_contig = mat_a.contiguous();
+  at::Tensor mat_b_contig = mat_b.contiguous();
+
+  cutlass::Status status;
+
+  if (out_dtype == at::ScalarType::BFloat16) {
+    using Config = Fp8GemmConfig<ElementInputFp8, cutlass::bfloat16_t>;
+    Fp8GemmRunner<typename Config::Gemm, cutlass::bfloat16_t> runner;
+    status = runner.run(mat_a_contig, mat_b_contig, scales_a_half, scales_b_half, out, hw_info);
```

Comment:
in sglang, you can only implement bfloat16. out data type is bfloat16 or float16.